extension/extensionlimiter/limiterhelper/consumer.go

```diff
@@ -0,0 +1,313 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package limiterhelper // import "go.opentelemetry.io/collector/extension/extensionlimiter/limiterhelper"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/xconsumer"
+	"go.opentelemetry.io/collector/extension/extensionlimiter"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// Consumer is a builder for creating wrapped consumers with resource limiters
+//
+// This supports limiting by request_count, request_items, and
+// memory_size weight keys.
+//
+// The network_bytes weight key not supported because that information
+// is not available from the pdata object.
+type Consumer struct {
+	requestItemsLimiter extensionlimiter.ResourceLimiter
+	memorySizeLimiter   extensionlimiter.ResourceLimiter
+	requestCountLimiter extensionlimiter.ResourceLimiter
+}
+
+// Config stores configuration from Options.
+type Config struct {
+	requestItemsLimiter bool
+	memorySizeLimiter   bool
+	requestCountLimiter bool
+}
+
+// Option represents the consumer options
+type Option func(*Config)
```

Comment:
We prefer the option of an interface with a private func for options.

Reply:
Fixed.